packages/aws-arch/scripts/common.ts

```diff
@@ -0,0 +1,47 @@
+/*********************************************************************************************************************
+ Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License").
+ You may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+/** Directory where generated files are output */
+export const GENERATED_DIR = path.resolve(__dirname, '..', 'src', 'generated');
+fs.ensureDirSync(GENERATED_DIR);
+
+export const CACHE_DIR = path.join(process.cwd(), 'node_modules', '.cache', '@aws-prototyping-sdk', 'aws-arch');
+fs.ensureDirSync(CACHE_DIR);
+
+export type TCountable = number | object | any[] | Set<any>;
+
+export function logCount(title: string, total: TCountable, count: TCountable, delta: boolean = false): void {
+	total = parseCount(total);
```

Comment:
`total` and `count` are local function parameters, mutating them does not mutate the source. Can create local variables to store the values we want to use in the function, but it would not change anything except take more memory for additional variables. Agree that limiting side-effects outside of a function is preferred, but you can look at a functions parameters as local variable within the function, and in no way cause side-effects by changing what they reference.